bullet_train-themes-tailwind_css/app/views/themes/tailwind_css/fields/_buttons.html.erb

```diff
@@ -1,23 +1,29 @@
 <%
 stimulus_controller = 'fields--button-toggle'
 form ||= current_fields_form
+
+# Initialize options.
```

Comment:
@gazayas I wonder if we can do some stuff here to be backwards compatible and to let people know that they should change things.

I'm thinking something like this, though I imagine this is incomplete:

```suggestion
# Initialize options.

# Start with a bit of handling for backwards compatibility
if options.is_a?(Array)
  ActiveSupport::Deprecation.warn(
    ":options as an Array is deprecated, use :button_field_options instead" \
    "The way to call 'shared/fields/buttons' has changed. See https://bullettrain.co/docs/TODO-WHAT-PAGE-DO-WE-LINK-TO"
  )
  button_field_options = options
  options = {} 
end
if html_options.is_a?(Hash)
  ActiveSupport::Deprecation.warn(
    ":html_options is deprecated, use :other_options instead" \
    "The way to call 'shared/fields/buttons' has changed. See https://bullettrain.co/docs/TODO-WHAT-PAGE-DO-WE-LINK-TO"
  )
  other_options = html_options
end
```

Reply:
@jagthedrummer
I agree with making the deprecation messages, and that we should explicitly write in the documentation about using the new keywords for the appropriate views. However, I think we should take a different approach on what we choose to deprecate (i.e. - `html_options`).

I know we most likely have differing views on this, but I'll go ahead and explain what I think about `html_options` to bring some clarity.

As the [documentation states](https://github.com/bullet-train-co/bullet_train-core/blob/622651936a4d0f570cd3920af5d64f5c523d561f/bullet_train/docs/field-partials/super-select.md?plain=1#L53) at least concerning Super Select, `html_options` is a key inherited from the underlying Rails form helper, and you can see this for different form helpers:

https://github.com/rails/rails/blob/23938052acd773fa24068debe56cd892cbf8d868/actionview/lib/action_view/helpers/form_options_helper.rb#L848-L850

https://github.com/rails/rails/blob/23938052acd773fa24068debe56cd892cbf8d868/actionview/lib/action_view/helpers/url_helper.rb#L234-L244

https://github.com/rails/rails/blob/23938052acd773fa24068debe56cd892cbf8d868/actionview/lib/action_view/helpers/url_helper.rb#L357

Also, from [the docs]():
> Because Bullet Train field partials have more responsibilities than the underlying Rails form field helpers, there are also additional options for things like hiding labels, displaying specific error messages, etc. For these options, we pass them separately as other_options. This keeps them separate from the options in options that will be passed directly to the underlying Rails form field helper.

For that reason, I don't think `other_options` should be affected in any way by this PR, so we shouldn't need it in the deprecation message.

So, now, I'm under the impression that we should define everything as follows:
1. `options`: In this case, options for buttons, for example `["one", "two", "three"]` (like it originally was).
2. `html_options`: The same as the Rails helpers in that we set the class attribute, etc.
3. `other_options`: Bullet Train specific options (i.e. - `help`, `hide_label`, `require`)

Concerning `html_options`, here's some information from the [Rails docs](https://api.rubyonrails.org/classes/ActionView/Helpers/UrlHelper.html):
> You can control the form and button behavior with html_options. Most values in html_options are passed through to the button element. For example, passing a :class option within html_options will set the class attribute of the button element.

So, what I'm mainly trying to say is, I think we should use these keywords in the same way that Rails originally uses them. I know that I've written `button_field_options` here, but now that I'm looking over things I think we should go with the 3 conventions I mentioned above.

I know there was much discussion on this in #499, and it's not a hill I'm willing to die on, I just wanted to share my thoughts on it before moving forward.

If my understanding of options is incorrect though, I'd be glad to go ahead and write the deprecations to reflect the code suggestion you added here.

Reply:
@gazayas I now understand what you were saying in that comment on #499 and I think you're right on. Both in that comment and in what you said above.

>I think we should use these keywords in the same way that Rails originally uses them.

💯 I couldn't agree more. Let's do that.

I think your breakdown of which thing means what is good and we should add it to the docs. Though... I do wonder if we could disambiguate `other_options` a little bit by calling it `bt_options` or `fp_options` or something with a little more significance and meaning than `other`.

Reply:
@jagthedrummer I definitely prefer `bt_options` or `fp_options`, I think it's definitely clearer. I admittedly wasn't too sure what all the differences were before, but I'm glad I took some time yesterday to sift through it. I'll try to make some progress here and ping you when I clean things up.